File: ruby/lib/Sikaru/types/case_input.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class CaseInput < Internal::Types::Model
      field :baseline, -> { Sikaru::Types::AnswerRef }, optional: false, nullable: false

      field :candidate, -> { Sikaru::Types::AnswerRef }, optional: false, nullable: false

      field :case_id, -> { String }, optional: false, nullable: false, api_name: "caseId"
    end
  end
end

File: python/src/sikaru_api/types/case_input.py

import typing

import pydantic
import typing_extensions
from ..core.pydantic_utilities import IS_PYDANTIC_V2, UniversalBaseModel
from ..core.serialization import FieldMetadata
from .answer_ref import AnswerRef


class CaseInput(UniversalBaseModel):
    baseline: AnswerRef
    candidate: AnswerRef
    case_id: typing_extensions.Annotated[str, FieldMetadata(alias="caseId"), pydantic.Field(alias="caseId")]

    if IS_PYDANTIC_V2:
        model_config: typing.ClassVar[pydantic.ConfigDict] = pydantic.ConfigDict(extra="allow", frozen=True)  # type: ignore # Pydantic v2
    else:

        class Config:
            frozen = True
            smart_union = True
            extra = pydantic.Extra.allow
